javascript/javascript2/week1/spiritAnimal.js

```diff
@@ -0,0 +1,38 @@
+const h3 = document.querySelector('h3')
+const btn = document.getElementById('myButton')
+const username = document.querySelector('#username')
+
+const withButton = document.getElementById('withbutton')
+const withHover = document.getElementById('withhover')
+const withText = document.getElementById('withtext')
+
+const spiritAnimals = ['The Laughing Butterfly', 'The Brave Badger', 'The Sharp Hawk', 'The Sleeper Rabbit', 'The Bully Tiger', 'The Blind Eagle', 'The Big Eye Wolf', 'The Runner Turtle', 'The Walking Grasshopper', 'The Lazy Ant']
+
+
+
+withText.addEventListener('click', function () {
+        username.addEventListener('keyup', function () {
+            const name = username.value
+            const random = Math.floor(Math.random() * 10)
+            h3.innerHTML = `${name} - ${spiritAnimals[random]}`
+        })
+})
+
+withHover.addEventListener('click', function () {
+        username.addEventListener('mouseenter', function () {
+            const name = username.value
+            const random = Math.floor(Math.random() * 10)
+            h3.innerHTML = `${name} - ${spiritAnimals[random]}`
+        })
+})
```

Comment:
None of these remove the other eventlisteners, so by the end of it, it changes on hover, click and input.

Reply:
Yes I was notice about that. I will try to find a solition for that and I will update my codes. Thanks 👍